src/devices/nice_flor_s.c

```diff
@@ -0,0 +1,83 @@
+/** @file
+    Nice Flor-s
+
+    Copyright (C) 2020 Samuel Tardieu <sam@rfc1149.net>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+/**
+Nice Flor-s
+
+Protocol description:
+The protocol has been analyzed at this link: http://phreakerclub.com/1615
+*/
+
+#include "decoder.h"
+
+static int nice_flor_s_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    if (bitbuffer->num_rows != 2 || bitbuffer->bits_per_row[1] != 0) {
+        return DECODE_ABORT_EARLY;
+    }
+    if (bitbuffer->bits_per_row[0] != 52) {
+        return DECODE_ABORT_LENGTH;
+    }
+
+    bitbuffer_invert(bitbuffer);
+    uint8_t *b = bitbuffer->bb[0];
+
+    uint8_t button_id = b[0] >> 4;
+    if (button_id < 1 || button_id > 4) {
+        return DECODE_ABORT_EARLY;
+    }
+    int repeat = (b[0] & 0xf) ^ 0xf ^ button_id;
+    char serial[8], code[5];
+    // Encrypted serial is made of nibbles 2 and 7 to 12
+    snprintf(serial, sizeof serial,
+             "%x%x%02x%02x%x",
+             b[1] >> 4, b[3] & 0xf, b[4], b[5], b[6] >> 4);
+    // Encrypted rolling code is made of nibbles 3 to 6
+    snprintf(code, sizeof code,
+             "%x%02x%x",
+             b[1] & 0xf, b[2], b[3] >> 4);
```

Comment:
can you use `int code =  ((b[1] & 0xf) << 12) | (b[2] << 4) | (b[3] >> 4);` and `DATA_FORMAT, "%04x", DATA_INT, code` below, also for serial?